rpcserver.go

```diff
@@ -1756,109 +1770,21 @@ out:
 func (r *rpcServer) OpenChannelSync(ctx context.Context,
 	in *lnrpc.OpenChannelRequest) (*lnrpc.ChannelPoint, error) {
 
-	rpcsLog.Tracef("[openchannel] request to NodeKey(%v) "+
-		"allocation(us=%v, them=%v)", in.NodePubkeyString,
-		in.LocalFundingAmount, in.PushSat)
-
-	// We don't allow new channels to be open while the server is still
-	// syncing, as otherwise we may not be able to obtain the relevant
-	// notifications.
-	if !r.server.Started() {
-		return nil, ErrServerNotActive
-	}
-
-	// Creation of channels before the wallet syncs up is currently
-	// disallowed.
-	isSynced, _, err := r.server.cc.wallet.IsSynced()
-	if err != nil {
+	if err := r.canOpenChannel(); err != nil {
 		return nil, err
 	}
-	if !isSynced {
-		return nil, errors.New("channels cannot be created before the " +
-			"wallet is fully synced")
-	}
 
-	// Decode the provided target node's public key, parsing it into a pub
-	// key object. For all sync call, byte slices are expected to be
-	// encoded as hex strings.
-	keyBytes, err := hex.DecodeString(in.NodePubkeyString)
```

Comment:
This field has been deprecated but was still being parsed, this commit changes that now. Do we want to return an error if the field is set?

Reply:
good catch! maybe we should continue to parse both because currently we don't parse `NodePubkey` via REST at all. that way we can give some more warning before switching over entirely.

fwiw this field is also printed in both `Trace` log statement, so the log in OpenChannel prob wasn't even that useful haha

Reply:
added parsing for `NodePubkey` in `OpenChannelSync` w/ fallback to `NodePubkeyString`, we can finish removing in a future version.